src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -329,9 +330,25 @@ private function getSchema(): Schema
         return $schema;
     }
 
-    public static function formatDateTime(\DateTimeInterface $dateTime)
+    /**
+     * @throws DBALException
+     *
+     * @deprecated since Symfony 4.4, use convertDateTime instead
+     */
+    public static function formatDateTime(\DateTimeInterface $dateTime, AbstractPlatform $platform = null)
```

Comment:
You've added a second parameter to the method, but I can find any calls with that parameter being set.

Reply:
See my commits in order and you will understand my logic :)

Reply:
that's a BC break, can't be done

Reply:
it is not a question of commits in order. The existing method that you want to deprecate must not change signature.

Reply:
you should remove the new argument and always use `$dateTime->format('Y-m-d\TH:i:s')` in this deprecated method. Your current change still requires changing the caller to add the platform to benefit from the fix, at which point it is as easy to switch to the new method instead.